src/main/java/org/openlmis/utils/BasicRequisitionDtoComparator.java

```diff
@@ -25,74 +25,62 @@
 import org.apache.commons.collections.comparators.ComparatorChain;
 import org.openlmis.requisition.dto.BasicRequisitionDto;
 import org.openlmis.requisition.exception.ValidationMessageException;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
 
-import java.util.Collections;
 import java.util.Comparator;
 import java.util.List;
 import java.util.Map;
 
 public class BasicRequisitionDtoComparator implements Comparator<BasicRequisitionDto> {
-  private static final Map<String, FieldComparator> AVAILABLE_COMPARATORS;
+  private static final Map<String, BeanComparator> AVAILABLE_COMPARATORS;
 
   static {
     AVAILABLE_COMPARATORS = Maps.newHashMap();
-    AVAILABLE_COMPARATORS.put("emergency", new FieldComparator("emergency", false));
-    AVAILABLE_COMPARATORS.put("programName",new FieldComparator("program.name"));
-    AVAILABLE_COMPARATORS.put("facilityCode", new FieldComparator("facility.code"));
-    AVAILABLE_COMPARATORS.put("facilityName", new FieldComparator("facility.name"));
+    AVAILABLE_COMPARATORS.put("emergency", new BeanComparator("emergency"));
+    AVAILABLE_COMPARATORS.put("programName", new BeanComparator("program.name"));
+    AVAILABLE_COMPARATORS.put("facilityCode", new BeanComparator("facility.code"));
+    AVAILABLE_COMPARATORS.put("facilityName", new BeanComparator("facility.name"));
   }
 
-  private List<String> compareConditions;
+  private List<Sort.Order> compareConditions;
 
   /**
-   * Creates new instance with a single compare condition.
+   * Creates new instance with the passed pageable instance. If pageable instance does not contain
+   * sort property (it is equal to null), the class will use default sort: emergency DESC and
+   * programName ASC.
    */
-  public BasicRequisitionDtoComparator(String compareCondition) {
-    this(Collections.singletonList(compareCondition));
-  }
+  public BasicRequisitionDtoComparator(Pageable pageable) {
+    Sort sort = pageable.getSort();
 
-  /**
-   * Creates new instance with the passed compare conditions.
-   */
-  public BasicRequisitionDtoComparator(List<String> compareConditions) {
-    this.compareConditions = isEmpty(compareConditions)
-        ? Lists.newArrayList("emergency", "programName")
-        : compareConditions;
+    if (null == sort) {
+      sort = new Sort(
+          new Sort.Order(ASC, "emergency"), new Sort.Order(ASC, "programName")
```

Comment:
The comment says about "emergency" descending order being default, but it is different here.